### DISCUSSION TITLE
`create-waku` from url

### DISCUSSION THREAD

**Discussion by himself65** [wakujs/waku#315] [Ideas]
support `create-waku --template <url>`

```
create-waku --template 01_counter // will download waku/${currentVersion}/examples/01_counter

create-waku --template himself65/waku-vercel-template // will download https://github.com/himself65/waku-vercel-template

create-waku --template https://github.com/himself65/waku-vercel-template // as-is
```

**Comment by dai-shi** [comment]
Sounds good. It would be as simple as just downloading but why not.

**Reply by himself65** [reply]
Oh, I was reading the create-vite and create-next cli, and I think the reason they have is that they have many options to create a project

**Reply by dai-shi** [reply]
I just meant why not we can do it. Meaning let's do it.

**Reply by himself65** [reply]
create-next-app is using https://www.npmjs.com/package/@vercel/ncc to bundle cli into one file. I'm also thing to use such way to create CLI

**Comment by ojj1123** [comment]
I've done to write the prototype for this feature. Could I PR?
I want to support this features:

```
pnpm create waku --example 14_react-tweet // will download https://github.com/dai-shi/waku/tree/main/examples/14_react-tweet

pnpm create waku --example https://github.com/dai-shi/waku/tree/main/examples/14_react-tweet

pnpm create waku --example https://github.com/himself65/waku-vercel-template
```

and I want to support `automatically installing dependencies` during setting up the project

**Reply by dai-shi** [reply]
If that works, I think we can drop `--choose-template` and only copy `01_template` in the package.

**Reply by ojj1123** [reply]
Okay i will

**Reply by dai-shi** [reply]
Just for fun, this is my initial version:
https://github.com/dai-shi/waku/blob/01cb0a64000c5cb8def83dd71eaa1d8bb943ae59/contrib/create-wakuwork/cli.js#L1-L12

**Reply by himself65** [reply]
https://github.com/dai-shi/waku/pull/581

**Reply by dai-shi** [reply]
> If that works, I think we can drop `--choose-template` and only copy `01_template` in the package.

I thought it would be nice, but it comes with a hack `DEFAULT_REF`.
That's my bad and #647 kind of reverts it.
